src/Shared/Microsoft.DotNet.Scaffolding.Helpers/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,148 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.DotNet.Scaffolding.Helpers.Extensions
+{
+    public static class StringExtensions
+    {
+        public static bool IsCSharpProject(this string projectFilePath)
+        {
+            return projectFilePath?.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase) == true;
+        }
+
+        public static bool IsBinary(this string filePath)
+        {
+            return
+                filePath?.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) == true ||
+                filePath?.EndsWith(".exe", StringComparison.OrdinalIgnoreCase) == true;
+        }
+
+        public static bool IsSolutionFile(this string filePath)
+        {
+            return
+                filePath?.EndsWith(".sln", StringComparison.OrdinalIgnoreCase) == true;
+        }
+
+        /// <summary>
+        /// since netstandard2.0 does not have a Contains that allows StringOrdinal param, using lower invariant comparison.
+        /// </summary>
+        /// <returns>true if lower invariants are equal, false otherwise. false for any null scenario.</returns>
+        public static bool ContainsIgnoreCase(this string input, string value)
```

Comment:
Yep, keeping these separate for this update since I didn't want the older (legacy scaffolders) to depend on Microsoft.DotNet.Scaffolding.Helpers yet. This project and the Microsoft.DotNet.Scaffolding.Shared will converge within the next few updates/previews. Will track [here](https://github.com/dotnet/Scaffolding/issues/2733)